src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1521,7 +1519,7 @@ private bool TryExtractValueFromAttribute<T>(CustomAttributeHandle handle, out T
                     // TODO: error checking offset in range
                     BlobReader reader = MetadataReader.GetBlobReader(valueBlob);
 
-                    if (reader.Length > 4)
+                    if (reader.Length >= 4)
```

Comment:
>if (reader.Length >= 4) [](start = 20, length = 23)

It is not clear what is the motivation for this change. Can the blob encode of 4 bytes long encode any values?

Reply:
This changes makes it so we succeed at decoding an ObsoleteAttribute with no constructor arguments and no named arguments.

Reply:
> This change makes it so we succeed at decoding an ObsoleteAttribute with no constructor arguments and no named arguments.

Why would we need this? This change affects decoding of all attributes across the board. Before we didn't even call ```valueExtractor``` in a situation like that. Now we do. This feels risky to me.

---
In reply to: [395187619](https://github.com/dotnet/roslyn/pull/42518#discussion_r395187619) [](ancestors = 395187619)

Reply:
I'll refactor to make this change unnecessary.